actionpack/lib/abstract_controller/helpers.rb

```diff
@@ -150,21 +150,14 @@ def modules_for_helpers(args)
           case arg
           when String, Symbol
             file_name = "#{arg.to_s.underscore}_helper"
-            begin
-              require_dependency(file_name)
-            rescue LoadError => e
-              raise AbstractController::Helpers::MissingHelperError.new(e, file_name)
-            end
-
             mod_name = file_name.camelize
             begin
               mod_name.constantize
-            rescue LoadError
-              # dependencies.rb gives a similar error message but its wording is
-              # not as clear because it mentions autoloading. To the user all it
-              # matters is that a helper module couldn't be loaded, autoloading
-              # is an internal mechanism that should not leak.
-              raise NameError, "Couldn't find #{mod_name}, expected it to be defined in helpers/#{file_name}.rb"
+            rescue NameError, LoadError => error
+              if error.message.match?(/Unable to autoload constant .*, expected .* to define it/i)
```

Comment:
I'm not super happy with that regexp match, but it's the only way I found to support both autoloaders.

Classic raises a `#<LoadError: Unable to autoload constant Admin::UsersHelper, expected /Users/byroot/src/github.com/Shopify/rails/actionpack/test/fixtures/helpers_typo/admin/users_helper.rb to define it>`

While Zeitwerk raises a `#<Zeitwerk::NameError: Unable to autoload constant Admin::UsersHelper, expected /Users/byroot/src/github.com/Shopify/rails/actionpack/test/fixtures/helpers_typo/admin/users_helper.rb to define it>`

Ideally I could segregate those with a `rescue LoadError, Zeitwerk::NameError`, however it would add a dependency on Zeitwerk in `actionpack`.